pkg/os/os.go

```diff
@@ -133,27 +134,9 @@ func (RealOS) LookupMount(path string) (containerdmount.Info, error) {
 // DeviceUUID gets device uuid of a device. The passed in device should be
 // an absolute path of the device.
 func (RealOS) DeviceUUID(device string) (string, error) {
-	const uuidDir = "/dev/disk/by-uuid"
-	if _, err := os.Stat(uuidDir); err != nil {
-		return "", err
-	}
-	files, err := ioutil.ReadDir(uuidDir)
+	output, err := exec.Command("/sbin/blkid", "-s", "UUID", "-o", "value", device).CombinedOutput()
```

Comment:
I tried an alpine container, and the output will be:
```console
# /sbin/blkid -s UUID -o value /dev/sda1
/dev/sda1: LABEL="cloudimg-rootfs" UUID="70045b4a-dd4c-4e37-889c-f1992da26c82" TYPE="ext4"
```
Did I misunderstand anything?

Reply:
Alpine uses `busybox` by default, you'll need `apk add -u blkid` first to get a full fat version.

Reply:
Cool. Tried it. Thanks

Reply:
But we may want to document the dependency in the README.md
Would you like to send another PR to update README.md?

Reply:
BTW, cadvisor is relying on `/dev/disk/by-uuid`, it means that cadvisor will not be available on linuxkit?

Reply:
Hrm, that would indeed be a problem but in the kubernetes source tree 9790389a11580d089be2efb6f93a1529e7270d6f:
```
$ git grep -lE 'by-(id|path|label|partuuid|uuid)' | grep -v _test.go$
cluster/gce/configure-vm.sh
cluster/gce/container-linux/configure-helper.sh
cluster/gce/gci/configure-helper.sh
examples/mysql-cinder-pd/README.md
federation/pkg/federation-controller/service/dns/dns.go
pkg/cloudprovider/providers/openstack/metadata.go
pkg/cloudprovider/providers/openstack/openstack_volumes.go
pkg/volume/aws_ebs/aws_util.go
pkg/volume/fc/fc_util.go
pkg/volume/gce_pd/gce_util.go
pkg/volume/iscsi/iscsi_util.go
pkg/volume/photon_pd/photon_util.go
pkg/volume/scaleio/sio_client.go
pkg/volume/util/device_util_linux.go
pkg/volume/vsphere_volume/vsphere_volume_util.go
test/kubemark/resources/start-kubemark-master.sh
vendor/github.com/codedellemc/goscaleio/volume.go
vendor/google.golang.org/api/compute/v0.beta/compute-api.json
vendor/google.golang.org/api/compute/v0.beta/compute-gen.go
vendor/google.golang.org/api/compute/v1/compute-api.json
vendor/google.golang.org/api/compute/v1/compute-gen.go
```
I think those are all particular to specific environments or helper scripts which aren't used by LinuxKit (at least not today).

The function you linked to is `getFsUUIDToDeviceNameMap` which isn't referenced in the kube tree and doesn't appear in `vendor/github.com/google/cadvisor/fs/fs.go` Might be an older cadvisor or a fork perhaps?

Reply:
Ah, I was on `release-1.7` I see the function you refer to in `master`.

I think I'll have to fix this on the LinuxKit side.